.github/actions/linux-build/action.yml

```diff
@@ -71,6 +55,21 @@ runs:
             sudo apt-get -qq -y install clang-14
           fi
         fi
+    - name: ccache 
+      id: ccache
+      uses: hendrikmuhs/ccache-action@v1.2
+      with:
+        # free total space for cache available on github is limited to 10GB, so only save on main branch;
+        # other branches will try to use cache from main, but branches that are too old won't benefit;
+        save: ${{ github.ref_name == 'main' }} 
+        # set enough space per build cache to keep a few previous versions to be used by older branches
+        # debug builds require more space
+        max-size: ${{ inputs.config == 'Release' && '200M' || '1G' }} 
+        # saves cache as `ccache-<key>-<timestamp>` (timestamp added automatically, key must be unique, github caches are never overwritten)
+        # key should be a composite of all options that affect the compilation
+        key: linux-${{ inputs.compiler }}-${{ inputs.version }}-${{ inputs.config }}-cov${{ inputs.coverage }}-dep${{ inputs.optional-dependencies }}-omp${{ inputs.openmp }}-san${{ inputs.sanitizers }}
```

Comment:
Can you add a small comment at the start of 'inputs:' as a reminder to change/update this key when any changes to the input are made?

Do we need to manage old caches after a change of keys, or are they deleted automatically at some point?

Reply:
good idea.

we don't need to manage caches, when the limit of 10G is reached, old caches are evicted automatically. New caches are never lost.